domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py

```diff
@@ -165,6 +166,10 @@ class EpMergeMachine(StateMachine):  # type: ignore[misc]
     def __init__(self, merge_job: MergeJob, main_event: Workflow = Workflow.PROCESS):
         self.job = merge_job
         self.main_event = main_event
+        self.cancellation_reason: str | None = None
+        self.original_tsoj: str | None = None
+        self.ep_metrics: list[DistributionMetric] = []
+        self.skipped_merge: bool = True
```

Comment:
Moving these to instance variables prevents issues with potential concurrent modification of cancellation_reason and original TSOJ.